src/plugins/saved_objects_finder/public/finder/saved_object_finder.tsx

```diff
@@ -0,0 +1,387 @@
+        col.push(item.defaultSearchField);
+      }
+      return col;
+    }, []);
+
+    const perPage = this.props.services.savedObjectsPlugin.settings.getListingLimit();
+    const response = await this.props.services.savedObjects.client.find<FinderAttributes>({
+      type: visibleTypes ?? Object.keys(metaDataMap),
+      fields: [...new Set(fields)],
+      search: queryText ? `${queryText}*` : undefined,
+      page: 1,
+      perPage,
+      searchFields: ['title^3', 'description', ...additionalSearchFields],
+      defaultSearchOperator: 'AND',
+      hasReference: this.props.services.savedObjectsManagement.getTagFindReferences({
+        selectedTags,
+        taggingApi: this.props.services.savedObjectsTagging,
+      }),
+    });
+
+    const savedObjects = response.savedObjects
+      .map((savedObject) => {
+        const {
+          attributes: { name, title },
+        } = savedObject;
+        const titleToUse = typeof title === 'string' ? title : '';
+        const nameToUse = name && typeof name === 'string' ? name : titleToUse;
+        return {
+          ...savedObject,
+          version: savedObject._version,
+          title: titleToUse,
+          name: nameToUse,
+          simple: savedObject,
+        };
+      })
+      .filter((savedObject) => {
+        const metaData = metaDataMap[savedObject.type];
+        if (metaData.showSavedObject) {
+          return metaData.showSavedObject(savedObject.simple);
+        } else {
+          return true;
+        }
+      });
+
+    if (!this.isComponentMounted) {
+      return;
+    }
+
+    // We need this check to handle the case where search results come back in a different
+    // order than they were sent out. Only load results for the most recent search.
+    if (query.text === this.state.query.text) {
+      this.setState({
+        isFetchingItems: false,
+        items: savedObjects,
+      });
+    }
+  }, 300);
+
+  constructor(props: SavedObjectFinderProps) {
+    super(props);
+
+    this.state = {
+      items: [],
+      isFetchingItems: false,
+      query: Query.parse(''),
+    };
+  }
+
+  public componentWillUnmount() {
+    this.isComponentMounted = false;
+    this.debouncedFetch.cancel();
+  }
+
+  public componentDidMount() {
+    this.isComponentMounted = true;
+    this.fetchItems();
+  }
+
+  private getSavedObjectMetaDataMap(): Record<string, SavedObjectMetaData> {
+    return this.props.savedObjectMetaData.reduce(
+      (map, metaData) => ({ ...map, [metaData.type]: metaData }),
+      {}
+    );
+  }
+
+  private fetchItems = () => {
+    this.setState(
+      {
+        isFetchingItems: true,
+      },
+      this.debouncedFetch.bind(null, this.state.query)
+    );
+  };
+
+  // server-side paging not supported
+  // 1) saved object client does not support sorting by title because title is only mapped as analyzed
+  // 2) can not search on anything other than title because all other fields are stored in opaque JSON strings,
```

Comment:
I think 2) is dated since we can search on other fields in the meantime (it's unrelated to this PR, because it was copied, but it just caught my eye)

Reply:
🙈 this is probably a **very** old comment

Reply:
I ended up just removing the comment since it wasn't really doing much anymore anyway